test/unit/text_layer_spec.js

```diff
@@ -90,4 +90,129 @@ describe("textLayer", function () {
+      container2 = document.createElement("div");
+    const waitCapability1 = Promise.withResolvers();
+
+    // Create text-content streams with dummy content.
+    const items1 = [
+      {
+        str: "Chapter A",
+        dir: "ltr",
+        width: 100,
+        height: 20,
+        transform: [20, 0, 0, 20, 45, 744],
+        fontName: "g_d0_f1",
+        hasEOL: false,
+      },
+      {
+        str: "page 1",
+        dir: "ltr",
+        width: 50,
+        height: 20,
+        transform: [20, 0, 0, 20, 45, 744],
+        fontName: "g_d0_f1",
+        hasEOL: false,
+      },
+    ];
+    const items2 = [
+      {
+        str: "Chapter B",
+        dir: "ltr",
+        width: 120,
+        height: 10,
+        transform: [10, 0, 0, 10, 492, 16],
+        fontName: "g_d0_f2",
+        hasEOL: false,
+      },
+      {
+        str: "page 2",
+        dir: "ltr",
+        width: 60,
+        height: 10,
+        transform: [10, 0, 0, 10, 492, 16],
+        fontName: "g_d0_f2",
+        hasEOL: false,
+      },
+    ];
+
+    const styles = {
+      g_d0_f1: {
+        ascent: 0.75,
+        descent: -0.25,
+        fontFamily: "serif",
+        vertical: false,
+      },
+      g_d0_f2: {
+        ascent: 0.5,
+        descent: -0.5,
+        fontFamily: "sans-serif",
+        vertical: false,
+      },
+    };
+    const lang = "en";
+
+    const streamGenerator1 = (async function* () {
+      for (const item of items1) {
+        yield { items: [item], styles, lang };
+        await waitCapability1.promise;
+      }
+    })();
+    const streamGenerator2 = (async function* () {
+      for (const item of items2) {
+        yield { items: [item], styles, lang };
+      }
+    })();
+
+    const textLayer1 = new TextLayer({
+      textContentSource: ReadableStream.from(streamGenerator1),
+      container: container1,
+      viewport: page1.getViewport({ scale: 1 }),
+    });
+    const textLayer1Promise = textLayer1.render();
+
+    const textLayer2 = new TextLayer({
+      textContentSource: ReadableStream.from(streamGenerator2),
+      container: container2,
+      viewport: page2.getViewport({ scale: 1 }),
+    });
+    await textLayer2.render();
+
+    // Ensure that the first textLayer has its rendering "paused" while
+    // the second textLayer renders.
+    waitCapability1.resolve();
+    await textLayer1Promise;
+
+    // Sanity check to make sure that all text was parsed.
+    expect(textLayer1.textContentItemsStr).toEqual(["Chapter A", "page 1"]);
+    expect(textLayer2.textContentItemsStr).toEqual(["Chapter B", "page 2"]);
+
+    const transform1 = Array.from(
+        container1.childNodes,
+        span => span.style.transform
+      ),
+      transform2 = Array.from(
+        container2.childNodes,
+        span => span.style.transform
+      );
+    expect(transform1).toEqual(["scaleX(1.20822)", "scaleX(0.947568)"]);
+    expect(transform2).toEqual(["scaleX(2.66371)", "scaleX(1.95972)"]);
```

Comment:
In hindsight this was obviously not a good idea, since the values will depend on *at least* the available fonts (which varies with e.g. OS) and the `devicePixelRatio`.